api/query/cache/backfill/backfill.go

```diff
@@ -126,31 +116,45 @@ func FillIndex(fetcher RunFetcher, logger shared.Logger, rt monitor.Runtime, int
 }
 
 func startBackfillMonitor(fetcher RunFetcher, logger shared.Logger, maxBytes uint64, m *backfillMonitor) error {
-	runs, err := fetcher.FetchRuns(int(maxBytes / bytesPerRun))
+	runsByProduct, err := fetcher.FetchRuns(int(maxBytes/bytesPerRun) / 4)
```

Comment:
Too much magic here. Explain a bit in the comment.

Also, what do we do when Edge is gone?

Reply:
Made it less magic (`len(shared.GetDefaultProducts())`, and commented.